packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.routing.js

```diff
@@ -55,6 +66,42 @@ export default /* @ngInject */ ($stateProvider) => {
         }
       },
 
+      checkoutOrderCartForUpgrade: /* @ngInject */ (
+        goBack,
+        goBackWithError,
+        isOptionFree,
+        $translate,
+        $window,
+        HostingCdnOrderService,
+      ) => async (autoPayWithPreferredPaymentMethod, addonPlan, serviceId) => {
+        try {
+          const {
+            order,
+          } = await HostingCdnOrderService.checkoutOrderCartForUpgrade(
+            isOptionFree || autoPayWithPreferredPaymentMethod,
```

Comment:
```suggestion
            autoPayWithPreferredPaymentMethod,
```
won't work if there are no payment mean

Reply:
isOptionFree it used in case where cdn is included so we should pass this as true.
eg: PERF offers

Reply:
If customer has no payment mean, we will try to autopay with default payment method which will fail so customer will never be able to order.

Reply:
fixed